### ISSUE TITLE
Hide 'OK' button in QuickPick Checkbox

### ISSUE THREAD

**Issue by jaeyonglee8687** [microsoft/vscode#143859]
Wondering there is any way to hide 'OK' button in quickPick checkbox, and echoing this stackOverflow question.
[https://stackoverflow.com/questions/58115365/how-to-remove-the-ok-button-from-quickpick](url).

Context: Teams toolkit uses the checkbox component during multiple steps, e.g., scaffolding an app requires several customer inputs, and one of them is choosing capabilities. 

<img width="899" alt="image" src="https://user-images.githubusercontent.com/87626977/155472926-75f170fe-d7d4-45be-a648-d30f82f8244e.png">

The quickPick title bar is consistently providing navigational actions (go back and move next), meanwhile, the checkbox component also shows a confirmation button together which duplicates the global navigation. I can imagine that the ok button is necessary when it's not in multiple steps, but not all the cases. 

Perhaps, can consider making it optional to display?

@misolori @daviddossett what do you think?

**Comment by misolori** [comment]
I think this one is a little tricky because providing this option does not guarantee that the extension author will provide a "next" step, where as here it's required. But I can see the desire to want to use the title bar nav for this instead and worry that it might not be too obvious how to proceed to the next step (which we could user test).

**Comment by daviddossett** [comment]
A couple of thoughts here:

## The OK button

This button being placed next to the input box doesn't feel at first glance like it serves as a confirmation/next step. I would have thought it was related to adding something to the list. 

While I'm at this step, it's worth calling out that the select all/none interaction here is a bit funky since it's really toggling between two different mixed selection states:

https://user-images.githubusercontent.com/25163139/155576479-64b2be04-72c3-411a-967d-47b090826733.mp4

## The checkmark

Input steps without multi-select don't need the checkmark to progress to the next step. I think you could remove the checkmark on these steps @jaeyonglee8687 

https://user-images.githubusercontent.com/25163139/155576929-e7f4d85b-ef69-45fb-b543-52c750824a32.mp4

I like the idea of providing a clearer way to move to the next step on a multi-select quick pick step though. Here's a quick example just poking around in dev tools. It could definitely be more obvious, but you get the idea:

<img width="668" alt="CleanShot 2022-02-24 at 09 38 46@2x" src="https://user-images.githubusercontent.com/25163139/155577688-52fa7b42-693d-47db-bfd2-ca55098d2c84.png">

## Progress indication

Lastly, I think it could be helpful to have an indicator of where the user is in the multi-step flow. It could be as simple as someting like this:

<img width="213" alt="CleanShot 2022-02-24 at 09 40 03@2x" src="https://user-images.githubusercontent.com/25163139/155578017-868874bf-69e1-433e-8619-e86781f7e77a.png">

**Comment by TylerLeonhardt** [comment]
Progress indication is supported via the following properties:

```ts
        const qp = window.createQuickPick();
	qp.title = 'Title';
	qp.step = 1;
	qp.totalSteps = 3;
	qp.items = [{ label: '1' }];
	qp.onDidAccept(() => {
		qp.step!++;
		qp.items = [{ label: qp.step + '' }];
	});
	qp.show();
```
![image](https://user-images.githubusercontent.com/2644648/155616502-db11e028-b608-4439-bb57-dab2358cacf1.png)

I asked:
> Sorry, are you contributing the ✔️ in the title bar? Do you do that for other steps that aren't a multi-select quick pick?

because I don't quite understand the use case of the ✔️ in the title bar. For non-multiselect quickpicks, I would expect that the thing that proceeds forward in the steps is selecting the item in the quick pick. Not hitting ✔️. And then for an input box, hitting ENTER is suppose to proceed.

In short, I feel like the ✔️ shouldn't be there and the user should use the `OK` instead for the multi-select... but I'm eager to understand your scenario.

**Comment by daviddossett** [comment]
Oh awesome, I didn't know that the progress indication was available in the API. Good stuff. It almost feels like that should be required if the quick pick has multiple steps? 

> For non-multiselect quickpicks, I would expect that the thing that proceeds forward in the steps is selecting the item in the quick pick. Not hitting ✔️. And then for an input box, hitting ENTER is suppose to proceed.

+1 on this

**Comment by jaeyonglee8687** [comment]
> Progress indication is supported via the following properties:
> 
> ```ts
>         const qp = window.createQuickPick();
> 	qp.title = 'Title';
> 	qp.step = 1;
> 	qp.totalSteps = 3;
> 	qp.items = [{ label: '1' }];
> 	qp.onDidAccept(() => {
> 		qp.step!++;
> 		qp.items = [{ label: qp.step + '' }];
> 	});
> 	qp.show();
> ```
> 
> ![image](https://user-images.githubusercontent.com/2644648/155616502-db11e028-b608-4439-bb57-dab2358cacf1.png)
> 
> I asked:
> 
> > Sorry, are you contributing the ✔️ in the title bar? Do you do that for other steps that aren't a multi-select quick pick?
> 
> because I don't quite understand the use case of the ✔️ in the title bar. For non-multiselect quickpicks, I would expect that the thing that proceeds forward in the steps is selecting the item in the quick pick. Not hitting ✔️. And then for an input box, hitting ENTER is suppose to proceed.
> 
> In short, I feel like the ✔️ shouldn't be there and the user should use the `OK` instead for the multi-select... but I'm eager to understand your scenario.

This makes sense to me. I was initially considering interactions without keyboard as well, and providing a consistent visual navigation on the title bar. However, as you said, selecting an option allows moving to the next, so the checkmark can be hidden. 
Wonder what would be the scenario enabling the checkmark though. 

@daviddossett  thanks for the feedback! The toggle behavior is not something I expected, will take a look more. For the progress indicator, it's hard to say how many steps are left because it dynamically changes according to the user selection in the previous stage. So we decided to remove it rather than confuse customers.

**Comment by jaeyonglee8687** [comment]
> <img alt="CleanShot 2022-02-24 at 09 38 46@2x" width="668" src="https://user-images.githubusercontent.com/25163139/155577688-52fa7b42-693d-47db-bfd2-ca55098d2c84.png">

I love this idea!

**Comment by jaeyonglee8687** [comment]
One particular exception I could think of is that a customer doesn't have access to a keyboard/enter key, how would proceed this (Well, this unlikely happens, but from an accessibility point of view, could happen).

![image](https://user-images.githubusercontent.com/87626977/155927233-87050c0c-015e-4dcd-a00a-bafc92a528af.png)

**Comment by misolori** [comment]
> a customer doesn't have access to a keyboard/enter key

Is this really a scenario? Even with our "touch" devices, there is always a keyboard that shows up so wondering how likely this actually happens vs a hypothetical scenario

For this specific example, if you are naming an object then it's best to display that as a quick pick options as well:

<img width="643" alt="CleanShot 2022-02-28 at 09 28 31@2x" src="https://user-images.githubusercontent.com/35271042/156029940-8b5b0775-593a-4212-a096-2cb08228e165.png">

**Comment by jaeyonglee8687** [comment]
Agree!
